kernel/scheduler.c

```diff
@@ -1296,6 +1296,8 @@ static __inl int pthread_destroy_task(struct task *t)
 {
     struct task *t_joiner;
 
+    /* delete from active threads of the process */
+    t->tb.threads[t->tb.tid - 1] = NULL;
```

Comment:
How can you be sure that this is the correct position in the array? positions in this array are reused. This looks wrong to me.